### PR TITLE
Apply `nixfmt` on many files

### PR DESCRIPTION
**Pull request by rycee**
Also add format checking in Travis CI job.

### CODE REVIEW

File: format [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  ! -path ./tests/modules/programs/gpg/override-defaults.nix \
+  ! -path ./tests/modules/programs/tmux/default.nix \
+  ! -path ./tests/modules/programs/tmux/disable-confirmation-prompt.nix \
+  ! -path ./tests/modules/programs/tmux/secure-socket-enabled.nix \
+  ! -path ./tests/modules/programs/zsh/session-variables.nix \
+  ! -path ./tests/modules/services/sxhkd/service.nix \
+  ! -path ./tests/modules/systemd/default.nix \
+  ! -path ./tests/modules/systemd/services.nix \
+  ! -path ./tests/modules/systemd/session-variables.nix \
+  -exec nixfmt $CHECK_ARG {} +
```
Comment (kalbasit) [inline format]: Why are the above paths ignored? Isn't it better to format all the files in one go?

Comment (rycee) [reply format]: They are touched by currently open PRs so should be left as-is to not cause unnecessary merge issues.

**Comment by rycee** [comment]
Oh, and about nixpkgs-fmt. I wasn't aware of it but looking at it now it does not seem to exactly solve the issue I'm having. Basically, I hoping to completely remove the need point out formatting issues in PRs since I find myself spending a disproportionate amount of time on such unnecessary matters.

**Comment by kalbasit** [comment]
Maybe @zimbatm can elaborate on the nix fmt tools. I'm just wondering what nixpkgs-fmt does not solve for you. I'm hoping to put in a formatter for shabka as well.

**Comment by zimbatm** [comment]
`nixpkgs-fmt` leaves a bit more freedom to the writer. It takes care of the indent and doesn't try to enforce things like line limits or collapse lines into one. It also tries to minimize the output diff that it would produce (no vertical alignments for example).

That being said the output should be readable by default. If you see issues with it let me know.

**Comment by sondr3** [comment]
I personally prefer the output from `nixfmt` over `nixpkgs-fmt`, almost entirely because of [this](https://github.com/nix-community/nixpkgs-fmt/issues/126) issue, it really rustles my jimmies. I still use `nixpkgs-fmt` to format my files because I know Rust and might be able to contribute, but I haven't found the time yet.

**Comment by zimbatm** [comment]
Makes sense. It doesn't conform with the style that's used in nixpkgs so it should be fixed.

**Comment by rycee** [comment]
I've tried out nixpkgs-fmt and do find it works quite well but I think it might be a better fit for Nixpkgs where more people are involved in reviewing pull requests. In HM it is mainly me that does reviews and I'd prefer a more aggressive tool that removes as much freedom as possible w.r.t formatting so that I can focus more on the PR's substance.

So I'll merge this for now but I'm not adverse to changing the tool in the future if other priorities arise.

**Pull request merged**